test/labeling/test_convergence.py

```diff
@@ -0,0 +1,93 @@
+import unittest
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from snorkel.analysis.utils import set_seed
+from snorkel.labeling.apply.pandas import PandasLFApplier
+from snorkel.labeling.lf import LabelingFunction, labeling_function
+from snorkel.labeling.model.label_model import LabelModel
+from snorkel.labeling.preprocess import preprocessor
+from snorkel.types import DataPoint
+
+
+class LabelingConvergenceTest(unittest.TestCase):
+    @classmethod
+    def setup_class(cls):
+        # Ensure deterministic runs
+        set_seed(123)
+
+        # Create raw data
+        cls.N_TRAIN = 1500
+        cls.N_VALID = 300
+
+        cls.k = 2
+        cls.df_train = create_data(cls.N_TRAIN)
+        cls.df_valid = create_data(cls.N_VALID)
+
+    @pytest.mark.complex
```

Comment:
How long does this take to run? See the instructions in CONTRIBUTING.md to measure.

Reply:
It's 0.94 seconds when I run the top 10 command in CONTRIBUTING.md. pytest takes around 1.5 seconds.